agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -1989,6 +1990,81 @@ bool backhaul_manager::handle_1905_topology_query(ieee1905_1::CmduMessageRx &cmd
+            bool front_iface = bool(i);
+            LOG(DEBUG) << "iterating on radio="
+                       << (front_iface ? soc->radio_mac : soc->radio_mac + " backhaul");
+
+            // Skip Backhaul iteration iface when STA BWL is not allocated (Eth connection or GW).
+            if (!front_iface && !soc->sta_wlan_hal) {
+                LOG(TRACE) << "Skip radio interface with no active STA BWL, front_radio="
+                           << soc->radio_mac;
+                continue;
+            }
+
+            auto localInterfaceInfo = tlvDeviceInformation->create_local_interface_list();
+
+            localInterfaceInfo->mac() =
+                front_iface ? network_utils::mac_from_string(soc->radio_mac)
+                            : network_utils::mac_from_string(soc->sta_wlan_hal->get_radio_mac());
+
+            LOG(DEBUG) << "Added radio interface to tlvDeviceInformation: "
+                       << localInterfaceInfo->mac();
+
+            ieee1905_1::eMediaType media_type = ieee1905_1::eMediaType::UNKNONWN_MEDIA;
+            if (soc->freq_type == beerocks::eFreqType::FREQ_24G) {
+                media_type = ieee1905_1::eMediaType::IEEE_802_11N_2_4_GHZ;
+            } else if (soc->freq_type == beerocks::eFreqType::FREQ_5G) {
+                media_type = ieee1905_1::eMediaType::IEEE_802_11AC_5_GHZ;
+            } else {
+                LOG(ERROR) << "Unsupported freq_type=" << int(soc->freq_type)
+                           << ", iface=" << soc->hostap_iface;
+                return false;
+            }
+            localInterfaceInfo->media_type() = media_type;
+
+            ieee1905_1::s802_11SpecificInformation media_info = {};
+            localInterfaceInfo->alloc_media_info(sizeof(media_info));
+
+            // BSSID field is not defined well for interface. The common definition is in simple
+            // words "the AP/ETH mac that we are connected to".
+            // For fronthaul radio interface or unused backhaul interface put zero mac.
+            if (local_gw || m_sConfig.eType == SBackhaulConfig::EType::Wired || front_iface ||
+                (m_sConfig.eType == SBackhaulConfig::EType::Wireless &&
+                 soc->sta_iface != m_sConfig.wireless_iface)) {
+                media_info.network_membership = network_utils::ZERO_MAC;
+            } else {
+                media_info.network_membership =
+                    network_utils::mac_from_string(soc->sta_wlan_hal->get_bssid());
+            }
+
+            // Note: Not sure that "NON_AP_NON_PCP_STA" is the correct definition to our backhaul
+            // interface.
+            media_info.role =
+                front_iface ? ieee1905_1::eRole::AP : ieee1905_1::eRole::NON_AP_NON_PCP_STA;
+
+            // TODO: The Backhaul manager does not hold the information on the front radios.
+            // For now, put zeros and when the Agent management will be move to unified Agent thread
+            // this field will be filled. #435
```

Comment:
Unify Agent is the entity that will connect to two fronthaul managers (per radio) that each one will have a monitor and ap manager. Meaning, it will be the main Agent process.
The backhaul manager will be responsible only to link establishment and remove from him all the Agent stuff.